ui/selectors/confirm-transaction.js

```diff
@@ -217,10 +218,18 @@ export const transactionFeeSelector = function (state, txData) {
   const currentCurrency = currentCurrencySelector(state);
   const conversionRate = conversionRateSelector(state);
   const nativeCurrency = getNativeCurrency(state);
+  const metaSwapAverageGasEstimate = getAveragePriceEstimateInHexWEI(state);
 
-  const {
-    txParams: { value = '0x0', gas: gasLimit = '0x0', gasPrice = '0x0' } = {},
+  const { txParams: { value = '0x0', gas: gasLimit = '0x0' } = {} } = txData;
+
+  // if the gas price from our infura endpoint is null or undefined
+  // use the metaswap average price estimation as a fallback
+  let {
```

Comment:
This change makes sense at face value but have we confirmed that there are no other places in the extension that Infura being down would effect, that don't use this selector?  I don't think I have the answer to this, but just wanted to make sure it was considered.

Reply:
Good question. I'll take dig into this.

Reply:
I'm not finding other places we can patch up right now and in any case I can't think of how it would cause any issues elsewhere to patch this by itself?